diff --git a/old/mac_addr.c b/new/mac_addr.c
index 5322a79..87e7065 100644
--- a/old/mac_addr.c
+++ b/new/mac_addr.c
@@ -97,10 +97,10 @@ esp_err_t esp_efuse_mac_get_custom(uint8_t *mac)
     uint8_t calc_crc = esp_rom_efuse_mac_address_crc8(mac, 6);
 
     if (efuse_crc != calc_crc) {
-        ESP_LOGE(TAG, "Base MAC address from BLK3 of EFUSE CRC error, efuse_crc = 0x%02x; calc_crc = 0x%02x", efuse_crc, calc_crc);
 #ifdef CONFIG_ESP_MAC_IGNORE_MAC_CRC_ERROR
         ESP_LOGW(TAG, "Ignore MAC CRC error");
 #else
+        ESP_LOGE(TAG, "Base MAC address from BLK3 of EFUSE CRC error, efuse_crc = 0x%02x; calc_crc = 0x%02x", efuse_crc, calc_crc);
         return ESP_ERR_INVALID_CRC;
 #endif
     }
@@ -137,10 +137,10 @@ esp_err_t esp_efuse_mac_get_default(uint8_t *mac)
         if (((mac_high & 0xFFFF) == 0x18fe) && (mac_low >= 0x346a85c7) && (mac_low <= 0x346a85f8)) {
             return ESP_OK;
         } else {
-            ESP_LOGE(TAG, "Base MAC address from BLK0 of EFUSE CRC error, efuse_crc = 0x%02x; calc_crc = 0x%02x", efuse_crc, calc_crc);
 #ifdef CONFIG_ESP_MAC_IGNORE_MAC_CRC_ERROR
             ESP_LOGW(TAG, "Ignore MAC CRC error");
 #else
+            ESP_LOGE(TAG, "Base MAC address from BLK0 of EFUSE CRC error, efuse_crc = 0x%02x; calc_crc = 0x%02x", efuse_crc, calc_crc);
             return ESP_ERR_INVALID_CRC;
 #endif
         }
